modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HorizontalCardTag.java

```diff
@@ -0,0 +1,383 @@
+				return false;
+			}
+
+			return _horizontalCard.isDisabled();
+		}
+
+		return _disabled;
+	}
+
+	public Boolean isSelectable() {
+		if (_selectable == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelectable();
+		}
+
+		return _selectable;
+	}
+
+	public Boolean isSelected() {
+		if (_selected == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelected();
+		}
+
+		return _selected;
+	}
+
+	public void setActionDropdownItems(List<DropdownItem> actionDropdownItems) {
+		_actionDropdownItems = actionDropdownItems;
+	}
+
+	public void setData(Map<String, String> data) {
+		_data = data;
+	}
+
+	public void setDisabled(Boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setHorizontalCard(HorizontalCard horizontalCard) {
+		_horizontalCard = horizontalCard;
+	}
+
+	public void setHref(String href) {
+		_href = href;
+	}
+
+	public void setIcon(String icon) {
+		_icon = icon;
+	}
+
+	public void setInteractive(Boolean interactive) {
+		_interactive = interactive;
+	}
+
+	public void setSelectable(Boolean selectable) {
+		_selectable = selectable;
+	}
+
+	public void setSelected(Boolean selected) {
+		_selected = selected;
+	}
+
+	public void setTitle(String title) {
+		_title = title;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_data = null;
+		_disabled = null;
+		_horizontalCard = null;
+		_href = null;
+		_icon = null;
+		_interactive = null;
+		_selectable = null;
+		_selected = null;
+		_title = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/HorizontalCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(
+		Map<String, Object> initialProps) {
+
+		Map<String, Object> props = super.prepareProps(initialProps);
+
+		props.put("actions", getActionDropdownItems());
+		props.put("cssClass", processCssClasses(new LinkedHashSet<>()));
+		props.put("disabled", isDisabled());
+		props.put("href", getHref());
+		props.put("icon", getIcon());
+		props.put("id", getId());
+		props.put("interactive", getInteractive());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("title", getTitle());
+
+		return props;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		if (isSelectable()) {
+			cssClasses.add("form-check");
+			cssClasses.add("form-check-card");
+			cssClasses.add("form-check-middle-left");
+		}
+		else {
+			cssClasses.add("card");
+			cssClasses.add("card-horizontal");
+		}
+
+		cssClasses.add("card-type-directory");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		Boolean selectable = isSelectable();
+
+		if ((getIcon() == null) && (getTitle() == null) && !selectable) {
+			return EVAL_BODY_INCLUDE;
```

Comment:
I have to admit, I don't fully understand what this does, I just copied it from previous tag class :sweat_smile:

If I recall correctly, somebody mentioned it on Slack that it is relevant when you chain tags. How exactly this relates to our cases, do we need to continue having the check, I am not sure.

Reply:
if `doStartTag` returns:

- EVAL_BODY_INCLUDE: Then it will print the content body of the tag (`<clay:card-horizontal>content body</clay:card-horizontal>) and then call `processEndTag`;
- SKIP_BODY: Then it will skip the rendering of the body and directly call to `processEndTag`.

You can read more about it in any reference for [Tag](https://tomcat.apache.org/tomcat-5.5-doc/jspapi/javax/servlet/jsp/tagext/Tag.html).

Reply:
I see. So, because we are not supporting content in a tag, something like:
```
<clay:horizontal-card>
    <div>Hello</div>
</clay:horizontal-card>
```
we don't need this, instead we are always skipping content (`return SKIP_BODY`).

Removing it :+1: